File: 2020/06/main.hs
import qualified Data.Set as Set
import Data.List.Split (splitOn)
import Data.List (lines)

handleGroup group = let sets = Set.fromList <$> group in
    foldl Set.intersection (head sets) (tail sets)

main = do
    groups <- (splitOn [""] . lines) <$> readFile "input"
    putStrLn . show . sum $ length . handleGroup <$> groups

File: 2020/06/main.py
#!/usr/bin/env python

from functools import reduce

groups = []

try:
    while True:
        sets = []
        while line := input():
            sets.append(set(line))

        try:
            groups.append(reduce(lambda done, s: done.intersection(s), sets))
        except TypeError:
            groups.append({})
except EOFError:
    try:
        groups.append(reduce(lambda done, s: done.intersection(s), sets))
    except TypeError:
        groups.append({})

sizes = [len(g) for g in groups]
print(sizes)
print(sum(sizes))
